web/src/buddy_list.ts

```diff
@@ -316,6 +316,7 @@ export class BuddyList extends BuddyListConf {
         $("#buddy-list-other-users-container .view-all-users-link").remove();
         if (!buddy_data.get_is_searching_users()) {
             this.render_view_user_list_links();
+            this.display_or_hide_sections();
```

Comment:
Could this result in header-hiding failing to update properly if `buddy_data.get_is_searching_users`? I guess maybe the theory is that when you search, you enter a defined headers state, and what is desired there never changes from that until search ends?

Would be good to have a comment in this block explaining the reasoning there.

Reply:
Yeah, the searching would only be a subset of what we would show in each section, or technically more complicated than that but we only hide sections if there's no "view more" link. I can add a comment.

Reply:
Oh dang, but users can switch the narrow when search is open... I guess we need to do it every time then!